hibernate-rx-core/src/main/java/org/hibernate/rx/impl/RxSessionInternalImpl.java

```diff
@@ -88,12 +123,46 @@ public Object immediateLoad(String entityName, Serializable id) throws Hibernate
 					} );
 		}
 		if ( association instanceof PersistentCollection ) {
-			//TODO: handle PersistentCollection (raise InitializeCollectionEvent)
-			throw new UnsupportedOperationException("fetch() is not yet implemented for collections");
+			PersistentCollection persistentCollection = (PersistentCollection) association;
+			return rxInitializeCollection( persistentCollection, false )
+					.thenApply( l -> Optional.ofNullable( (T) l ) );
 		}
 		return RxUtil.completedFuture( Optional.ofNullable(association) );
 	}
 
+	/**
+	 * @deprecated use {@link #rxInitializeCollection(PersistentCollection, boolean)} instead
+	 */
+	@Deprecated
+	@Override
+	public void initializeCollection(PersistentCollection collection, boolean writing) {
+		throw getExceptionConverter().convert( new UnsupportedOperationException( "RxSessionInternalImpl#initializeCollection not supported, use rxInitializeCollection instead" ) );
+	}
+
+	public <T> CompletionStage<T> rxInitializeCollection(PersistentCollection collection, boolean writing) {
+		checkOpenOrWaitingForAutoClose();
+		pulseTransactionCoordinator();
+		InitializeCollectionEvent event = new InitializeCollectionEvent( collection, this );
+		return fire( event, EventType.INIT_COLLECTION, (DefaultRxInitializeCollectionEventListener l) -> l::onRxInitializeCollection )
+				.handle( (v, e) -> {
+					delayedAfterCompletion();
+
+					if ( e instanceof MappingException ) {
+						throw getExceptionConverter().convert( new IllegalArgumentException( e.getMessage() ) );
+					}
+					else if ( e instanceof RuntimeException ) {
+						throw getExceptionConverter().convert( (RuntimeException) e );
+					}
+					else if ( e != null ) {
+						return RxUtil.rethrow( e );
+					}
+					return v;
```

Comment:
I think we have a bunch of places where we do something quite similar to this. I need to review that and see if we can just factor out a util function. (But of course that doesn't affect this PR.)

Reply:
It think there is already a method in RxUtil that does that.

Reply:
No, but I mean the whole flow of exception conversion. Don't worry, I'll have a look and see if we really have duplication or if they're all subtly different.

Reply:
Oh well, gee, there are a *lot* of places which do very-similar looking conversion, but some of them are subtly different (differences that are inherited from core `SessionImpl`) so I'm scared to touch this.